source/module_hamilt_general/module_vdw/test/vdw_test.cpp

```diff
@@ -0,0 +1,607 @@
+#include "module_cell/unitcell.h"
+#include "module_cell/setup_nonlocal.h"
+#include "module_io/input.h"
+#include "module_base/mathzone.h"
+#include "module_base/vector3.h"
+#include"gtest/gtest.h"
+#include"gmock/gmock.h"
+#include "mpi.h"
+#define private public
+#include "module_hamilt_general/module_vdw/vdwd2_parameters.h"
+#include "module_hamilt_general/module_vdw/vdwd3_parameters.h"
+#include "module_hamilt_general/module_vdw/vdwd2.h"
+#include "module_hamilt_general/module_vdw/vdwd3.h"
+#include "module_hamilt_general/module_vdw/vdw.h"
+
+
+/************************************************
+*  unit test of class VDW and related functions
+***********************************************/
+
+/**
+* - Tested functions:
+*   - make_vdw:
+*       Based on the value of INPUT.vdw_method, construct 
+*       Vdwd2 or Vdwd3 class, and do the initialization.
+*   - 
```

Comment:
Please add full annotations for your UT s to facilitate future review of the related codes.